test/unit/test_host.rb

```diff
@@ -65,7 +65,6 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com') == Host.new('example.com')
```

Comment:
The rubocop cop for detecting what they call "useless comparison" is pretty naive: https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/lint/useless_comparison.rb

It is only checking if the receiver of `#==` is the same as the arguments passed in.

I agree that since `Object#==` is being overwritten by an alias that there should be a test, however I think this is an example where the test is testing too many things (not just because of the 3 asserts, in my opinion it's fine to have all of the asserts in one test when dealing with aliases).

What this is really testing is that the initializer is deterministic given the same input (i.e. subsequent calls to the same initializer multiple times does not have different side effects) but actually more specifically that the implementation of the comparison depends only on the deterministic aspects of the initializer.

My personal style of testing would lead me to use stubs to test only what I'm trying to test, in this case that two hosts are equal when their hashes are equal.  I went ahead and opened #276 that adds the test for `==` back but does it in a way that appeases rubocop and only tests the equality method(s).